File: src/main/java/com/duxtinto/intellij/plugin/github/codereviews/domain/issues/IssuesDomainContract.java
package com.duxtinto.intellij.plugin.github.codereviews.domain.issues;

import org.jetbrains.annotations.Nullable;

public interface IssuesDomainContract {
    interface Switcher {
        void switchToIssue(IssueEntity issue);
    }
    interface Fetcher {
        @Nullable
        IssueEntity fetchIssueById(long id);
    }
}

File: src/main/kotlin/com/duxtinto/intellij/plugin/github/codereviews/domain/issues/IssuesDomainContract.kt
package com.duxtinto.intellij.plugin.github.codereviews.domain.issues

interface IssuesDomainContract {
    interface Switcher {
        fun switchToIssue(issue: IssueEntity)
    }

    interface Fetcher {
        fun fetchIssueById(id: Long): IssueEntity?
    }
}
